test/unit/runner.test.js

```diff
@@ -152,20 +157,40 @@ describe('runner', function () {
             });
         });
 
-        it('should filter down to the provided groupId correctly', function (done) {
-            Runner.extractRunnableItems(collection, 'F1', function (err, runnableItems, entrypoint) {
+        it('should filter item correctly by name', function (done) {
```

Comment:
Also add a test for folder-request name/id collisions.

Reply:
This PR does not handle name/id collisions.